main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -848,7 +848,6 @@ object Weeder2 {
     }
 
     def visitExpr(exprTree: Tree)(implicit sctx: SharedContext): Validation[Expr, CompilationMessage] = {
-      assert(exprTree.kind == TreeKind.Expr.Expr)
```

Comment:
why do you remove this line?

Reply:
I think we should keep this line.

Reply:
No because now `visitExpr` can accept trees that do not have this kind.